classes/Task/Update/UpdateFiles.php

```diff
@@ -184,6 +184,17 @@ protected function upgradeThisFile($orig): bool
 
                 return true;
             } catch (IOException $e) {
+                $checksumOrig = md5_file($orig);
+                $checksumDest = md5_file($dest);
+
+                $isFileUnchanged = $checksumOrig === $checksumDest;
+
+                if ($isFileUnchanged) {
+                    $this->logger->warning($this->translator->trans('Unable to copy %1$s – source and destination appear identical, skipping.', [$file]));
```

Comment:
This wording makes me think we won't even try to copy if the file is the same. Here is another version:

```suggestion
                    $this->logger->warning($this->translator->trans('Unable to copy %s, but ignoring as source and destination appear identical.', [$file]));
```